package-examples/tenant/ns-invariant.yaml

```diff
@@ -0,0 +1,15 @@
+apiVersion: fn.kpt.dev/v1alpha1
+kind: StarlarkRun
+metadata:
+  name: ns-invariant
+source: |
+  def ensure_one_namespace(resource_list):
```

Comment:
super minor suggestion but could be cool to have the source in a separate file so one can get the syntax highlighting.  Totally optional.

Reply:
That's a great idea actually. We should be able to have a just the .py file separately so that users can use IDE and also test the scripts. But it might complicate the referencing a bit, Kptfile->starlark-config-yaml->.py file. Instead, we can accept configMap option for starlark function in order to accept the source .py file directly.

cc @mengqiy @droot

Reply:
It thought starlark was .bzl (https://docs.bazel.build/versions/main/ide.html#visual-studio-code) but either way is great.  I think the configmap for starlark would be awesome and just take any entry in there as a file name and file path, that way more complex logic can be broken down into several files.

Reply:
include-file proposal here https://github.com/GoogleContainerTools/kpt/issues/2350 addresses exactly what you guys are asking. We can probably address this as part of starlark enhancement this quarter /cc @mengqiy